lib/linguist/languages.yml

```diff
@@ -1099,6 +1099,13 @@ DTrace:
   codemirror_mode: clike
   codemirror_mime_type: text/x-csrc
   language_id: 85
+Dafny:
+  type: programming
+  extensions:
+  - ".dfy"
```

Comment:
According to the [project's wiki](https://github.com/dafny-lang/dafny/wiki/Cautionary-Tales#understand-dafnys-command-line-interface), there's an interpreter named `dafny`. This should be added under `interpreters`.

Add this:

~~~diff
 Dafny:
   type: programming
   extensions:
   - ".dfy"
+  interpreters:
+  - dafny
   tm_scope: text.dfy.dafny
~~~

Reply:
That seems reasonable. It isn't technically an interpreter since it cross-compiles Dafny to several other target languages instead, but who's counting. :)

What does linguist do with this field exactly?

Reply:
I should say that if you pass the right flag to `dafny` it will compile and then run your program, so it can be used like an interpreter. If the point of this field is to allow the classifier to infer language from shebang lines, then that's valid, even though it's unlikely anyone would use Dafny that way.

Reply:
> If the point of this field is to allow the classifier to infer language from shebang lines

Yep, that's pretty much it. 👍